pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -347,49 +406,40 @@ func (m *ManagerImpl) addEndpoint(r *pluginapi.RegisterRequest) {
 			device.Health = pluginapi.Unhealthy
 			devices[device.ID] = device
 		}
-		existingDevs = devices
-	}
-	m.mutex.Unlock()
-
-	socketPath := filepath.Join(m.socketdir, r.Endpoint)
-	e, err := newEndpointImpl(socketPath, r.ResourceName, existingDevs, m.callback)
-	if err != nil {
-		glog.Errorf("Failed to dial device plugin with request %v: %v", r, err)
-		return
-	}
-	m.mutex.Lock()
-	if r.Options != nil {
-		m.pluginOpts[r.ResourceName] = r.Options
-	}
-	// Check for potential re-registration during the initialization of new endpoint,
-	// and skip updating if re-registration happens.
-	// TODO: simplify the part once we have a better way to handle registered devices
-	ext := m.endpoints[r.ResourceName]
-	if ext != old {
-		glog.Warningf("Some other endpoint %v is added while endpoint %v is initialized", ext, e)
-		m.mutex.Unlock()
-		e.stop()
-		return
+		e.devices = devices
```

Comment:
@vikaschoudhary16 @RenaudWasTaken  @jiayingz i believe endpoint don't have to cache devices, 
so i create a pull request to simplify device manager #65948, please take a look, i hope it makes sense.

Reply:
we might need cached devices for resource classes/compute resources. I would suggest to hold on removing cached devices from endpoint for some time.